hydromt/components/mesh.py

```diff
@@ -1,40 +1,384 @@
+
+        Parameters
+        ----------
+        region : dict
+            Dictionary describing region of interest, bounds can be provided for type 'mesh'.
+            In case of 'mesh', if the file includes several grids, the specific 2D grid can
+            be selected using the 'grid_name' argument.
+            CRS for 'bbox' and 'bounds' should be 4326; e.g.:
+
+            * {'bbox': [xmin, ymin, xmax, ymax]}
+
+            * {'geom': 'path/to/polygon_geometry'}
+
+            * {'mesh': 'path/to/2dmesh_file'}
+
+            * {'mesh': 'path/to/mesh_file', 'grid_name': 'mesh2d', 'bounds': [xmin, ymin, xmax, ymax]}
+        res: float
+            Resolution used to generate 2D mesh [unit of the CRS], required if region
+            is not based on 'mesh'.
+        crs : EPSG code, int, optional
+            Optional EPSG code of the model or "utm" to let hydromt find the closest projected CRS.
+            If None using the one from region, and else 4326.
+        grid_name : str, optional
+            Name of the 2D grid in mesh, by default "mesh2d".
+
+        Returns
+        -------
+        mesh2d : xu.UgridDataset
+            Generated mesh2d.
+
+        """  # noqa: E501
+        self._logger.info("Preparing 2D mesh.")
+
+        # Create mesh2d
+        mesh2d = create_mesh2d(
+            region=region,
+            res=res,
+            crs=crs,
+            logger=self._logger,
+        )
+        # Add mesh2d to self.mesh
+        self.set(mesh2d, grid_name=grid_name)
+
+        # This setup method returns mesh2d so that it can be wrapped for models
+        # which require more information
+        return mesh2d
+
+    @property
+    def data(self) -> Union[xu.UgridDataArray, xu.UgridDataset]:
+        """
+        Model static mesh data. It returns a xugrid.UgridDataset.
+
+        Mesh can contain several grids (1D, 2D, 3D) defined according
+        to UGRID conventions. To extract a specific grid, use get_mesh
+        method.
+        """
+        # XU grid data type Xarray dataset with xu sampling.
+        if self._data is None:
+            self._initialize_mesh()
+        return self._data
+
+    def _initialize_mesh(self, skip_read: bool = False) -> None:
+        """Initialize mesh object."""
+        if self._data is None:
+            self._data = xu.UgridDataset(xr.Dataset())
+            if self._root.is_reading_mode() and not skip_read:
+                self.read()
+
+    @property
+    def crs(self) -> Optional[CRS]:
+        """Returns model mesh crs."""
+        if self.data is not None:
+            for _, v in self.data.ugrid.crs.items():
+                return v
+        return None
+
+    @property
+    def bounds(self) -> Optional[Dict]:
+        """Returns model mesh bounds."""
+        if self.data is not None:
+            return self.data.ugrid.bounds
+        return None
+
+    @property
+    def mesh_region(self) -> Optional[gpd.GeoDataFrame]:
```

Comment:
I think this will be named region again in @deltamarnix PR on regions so maybe you could change the name already or check with him how to best update?